specifyweb/frontend/js_src/lib/components/Preferences/UserDefinitions.tsx

```diff
@@ -1347,6 +1347,30 @@ export const userPreferenceDefinitions = {
             defaultValue: false,
             type: 'java.lang.Boolean',
           }),
+          orderByField: definePref<'fullName' | 'name' | 'nodeNumber' | 'rankId'>({
```

Comment:
sorting by rankId first seems like a great idea
what are the use cases when that is not desirable?
e.g maybe we don't need a pref, and sorting by rankId first and name 2nd simply makes sense in all cases?

Reply:
and if this is not a pref, then there is no reason not to do sorting on the back-end

Reply:
I wanted to keep it as a preference in case users prefer the current behavior, but I don't have any examples where this behavior would be better. I know some are in favor of using alphabetical sorting before all else, so I worry this would disrupt someone's workflow. Thinking of https://xkcd.com/1172/

Using `rankId` as the default is likely to be more useful in most contexts. I would ask a developer to add support on the back end, but I am working on this outside of my regular duties, and we have important deadlines approaching. 

Thank you for reviewing this!

Reply:
> Thinking of https://xkcd.com/1172/

haha, love it!

at Esri, there is a policy that you cannot make a breaking change unless you deprecated the old behavior and waited for 3 releases to give people time to adapt (3 releases === 1 year). Having to maintain both old and new way for one year means there is a high bar before any new feature is added. So I have grown more conservative with features and customizations I add 😆